cmd/misc/main.go

```diff
@@ -951,53 +954,70 @@ func indexMissingBlocks(start uint64, end uint64, bt *db.Bigtable, client *rpc.E
 		end = uint64(lastBlockFromBlocksTable)
 	}
 
+	errFields := map[string]interface{}{
+		"start": start,
+		"end":   end}
+
 	batchSize := uint64(10000)
-	if start == 0 {
-		start = 1
-	}
-	for i := start; i < end; i += batchSize {
+	for from := start; from <= end; from += batchSize {
 		targetCount := batchSize
-		if i+targetCount >= end {
-			targetCount = end - i
+		if from+targetCount >= end {
+			targetCount = end - from + 1
 		}
-		to := i + targetCount - 1
+		to := from + targetCount - 1
 
-		list, err := bt.GetBlocksDescending(uint64(to), uint64(targetCount))
+		errFields["from"] = from
+		errFields["to"] = to
+		errFields["targetCount"] = targetCount
+
+		list, err := bt.GetBlocksDescending(to, targetCount)
 		if err != nil {
-			utils.LogError(err, "can not retrieve blocks via GetBlocksDescending from bigtable", 0)
+			utils.LogError(err, "error retrieving blocks from tableData", 0, errFields)
 			return
 		}
-		if uint64(len(list)) == targetCount {
-			logrus.Infof("found all blocks [%v]->[%v]", i, to)
-		} else {
-			logrus.Infof("oh no we are missing some blocks [%v]->[%v]", i, to)
-			blocksMap := make(map[uint64]bool)
-			for _, item := range list {
-				blocksMap[item.Number] = true
+
+		receivedLen := uint64(len(list))
+		if receivedLen == targetCount {
+			logrus.Infof("found all blocks [%v]->[%v], skipping batch", from, to)
+			continue
+		}
+
+		logrus.Infof("%v blocks are missing from [%v]->[%v]", targetCount-receivedLen, from, to)
+
+		blocksMap := make(map[uint64]bool)
+		for _, item := range list {
+			blocksMap[item.Number] = true
+		}
+
+		for block := from; block <= to; block++ {
+			if blocksMap[block] {
+				// block already saved, skip
+				continue
```

Comment:
```suggestion
		for block := from; block <= to && !blocksMap[block] ; block++ {
```
is shorter but maybe less readable

Reply:
You are right, this is really less readable :)
But honestly, it is a matter of taste and I think readability tastes way better than less lines ... something like that. Won't change.